plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/GlueTableOperations.java

```
@@ -0,0 +1,236 @@
+
+            String metadataLocation = table.getParameters().get(METADATA_LOCATION);
+            if (metadataLocation == null) {
+                throw new TrinoException(ICEBERG_INVALID_METADATA, format("Table is missing [%s] property: %s", METADATA_LOCATION, getSchemaTableName()));
+            }
+            return metadataLocation;
+        });
+    }
+
+    @Override
+    protected void commitNewTable(TableMetadata metadata)
+    {
+        String newMetadataLocation = writeNewMetadata(metadata, version + 1);
+        Map<String, String> parameters = ImmutableMap.of(
+                TABLE_TYPE_PROP, ICEBERG_TABLE_TYPE_VALUE.toUpperCase(ENGLISH),
+                METADATA_LOCATION, newMetadataLocation);
+        TableInput tableInput = new TableInput()
+                .withName(tableName)
+                .withTableType(EXTERNAL_TABLE.name())
+                .withOwner(owner.orElse(null))
+                .withParameters(parameters);
+
+        boolean succeeded = false;
+        try {
+            stats.getCreateTable().call(() -> {
+                glueClient.createTable(new CreateTableRequest()
+                        .withCatalogId(catalogId)
+                        .withDatabaseName(database)
+                        .withTableInput(tableInput));
+                return null;
+            });
+            succeeded = true;
+        }
+        catch (ConcurrentModificationException e) {
+            throw new TrinoException(ICEBERG_COMMIT_ERROR, format("Cannot commit %s because Glue detected concurrent update", getSchemaTableName()), e);
+        }
+        catch (AlreadyExistsException e) {
+            throw new TableAlreadyExistsException(getSchemaTableName());
+        }
+        catch (RuntimeException e) {
+            throw new TrinoException(ICEBERG_COMMIT_ERROR, format("Cannot commit %s due to unexpected exception", getSchemaTableName()), e);
+        }
+        finally {
+            cleanupMetadataLocation(!succeeded, newMetadataLocation);
```

this can swallow exception in flight.